### DISCUSSION TITLE
Create a new project without automatically installing Laravel/Sanctum

### DISCUSSION THREAD

**Discussion by benhdev** [laravel/framework#42381] [Ideas] [answered]
I find myself often writing my own authentication guards without using Laravel/Sanctum (don't get me wrong, its an awesome package) however, I just enjoy making my own a lot of the time.

It often frustrates me how every new project I make I have to manually remove the Sanctum composer package and all files related and edit the models to work without it. I'd appreciate a way to create a laravel project without Sanctum being automatically installed (since its not required functionally, it would make sense)

Adding to my last point, since Sanctum is not actually required functionally to run Laravel or authenticate users, it doesn't make sense to include this package in a fresh install every time. While it may save some users a lot of time, I can imagine a lot of users do also make their own authentication systems and so the installer should include a way to remove packages that aren't required. Or perhaps, the installer by default should create a project WITHOUT the packages, and then an option is provided if you want this package enabled.

**Comment by benhdev** [comment, marked as answer]
Decided i'd make my own template repository since this probably wont change anything

feel free to check it out [https://github.com/benhdev/laravel-skeleton](https://github.com/benhdev/laravel-skeleton)

**Reply by madalinignisca** [reply]
Myself, I did the same, for me and my team. I looked over your approach, but you went and made a skeleton fitting with a lots extras you considered.

I opened the discussion to highlight the need of a LIGHT skeleton, similar to what Symfony offers as minimalist to start and API or console only app. Considering something like that, adding either Passport of Sanctum would be a require of the package and almost done, especially by automating publishing their assets (config, scaffolding code etc.).

PS: Why don't you use redis extension and fallback to the slow predis implementation for redis?

**Reply by madalinignisca** [reply]
PS: Same day you opened a subject on the same topic I did: https://github.com/laravel/framework/discussions/42373

**Reply by benhdev** [reply]
Yeah, i started making the skeleton but then I also thought why not add some extras that I feel like I'm going to use. I've not updated it for a few days now and there's quite a few changes... I also started making my own framework too, just out of curiosity and as a learning experience/weekend boredom 😂 

I'm going to reach out, perhaps we can discuss this further and maybe talk about some ideas.

**Reply by madalinignisca** [reply]
Let's talk.
